tfe/resource_tfe_notification_configuration.go

```diff
@@ -0,0 +1,205 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceTFENotificationConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFENotificationConfigurationCreate,
+		Read:   resourceTFENotificationConfigurationRead,
+		Update: resourceTFENotificationConfigurationUpdate,
+		Delete: resourceTFENotificationConfigurationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"destination_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						string(tfe.NotificationDestinationTypeGeneric),
+						string(tfe.NotificationDestinationTypeSlack),
+					},
+					false,
+				),
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"token": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Sensitive: true,
+			},
+
+			"triggers": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice(
```

Comment:
@vancluever, yep, this worked :) In addition to passing the tests I wrote, it also stopped me from putting in various invalid strings when I was creating notifications with Terraform locally.

Reply:
Weird. I'm asking @radeksimko to weigh in here as I'm not too sure how this is working given the restrictions in place. Maybe the acceptance testing suite is not using `InternalValidate` after all.

Reply:
This works just fine because you are validating each element, not the whole set. In other words the `ValidateFunc` is associated with `TypeString`, not `TypeSet`.

Reply:
@radeksimko what I'm wondering about is how it's passing `InternalValidate` with the current code in place: https://github.com/hashicorp/terraform/blob/v0.12.6/helper/schema/schema.go#L831-L836